### DISCUSSION TITLE
How to hide popover on click inside popover?

### DISCUSSION THREAD

**Discussion by Rusinas** [Akryum/v-tooltip#420] [Q&A]
I need to close popover after user clicks on some item in popover (e.g. site navigation)

**Comment by kai-yu** [comment]
Cannot you use directive `v-close-popover` on the desire node?

**Comment by nonsocode** [comment]
what if you have a custom component in the popover? how do you close the popover from within that component?

I propose that the popover slot be a scoped slot with a handle to the close function

**Comment by DimosPagakis** [comment]
@Rusinas @nonsocode 
I am using a custom component inside the popover and I got it to work by doing the following.

Used the open property for the popover. Defaults to false so the popover is not open when the page is loaded
```<v-popover :open="openGuests" offset="16" placement="bottom-start">```

I added a custom trigger to open the popover by changing my boolean variable to true
```<p @click="openGuests = true">```

Inside the custom component that shows up in the popover, I emit an event to its parent.
```
click(){
 ... handle the click
 this.$emit('save')
}
```
Finally, on the page that I use the popover I handle the emitted event and set the boolean variable to false again.
```<guests-filter @save="openGuests = false"></guests-filter>```

I tested it and it works for multiple components I use inside the popovers

**Comment by DimosPagakis** [comment]
Well in this case I believe you could create a component that includes both
the popover and the "button" that toggles it. That way you only have one
variable and you just pass in props that make each component unique in the
way you desire
